File: backend/src/main/java/com/apocalypse/thefall/entity/TerrainConfiguration.java
package com.apocalypse.thefall.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.Map;

@Entity
@Getter
@Setter
@Table(name = "terrain_configuration")
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TerrainConfiguration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;

    @Column(columnDefinition = "jsonb", nullable = false)
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, String> descriptions;

    @Column(nullable = false)
    private int movementCost;

    @Column(nullable = false)
    private boolean walkable = true;

    @Column(name = "path", nullable = false)
    private String path;
}

File: backend/src/main/java/com/apocalypse/thefall/entity/TerrainConfiguration.kt
package com.apocalypse.thefall.entity

import jakarta.persistence.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes

@Entity
@Table(name = "terrain_configuration")
open class TerrainConfiguration {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(unique = true, nullable = false)
    open var name: String = ""

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb", nullable = false)
    open var descriptions: MutableMap<String, String> = mutableMapOf()

    @Column(nullable = false)
    open var movementCost: Int = 0

    @Column(nullable = false)
    open var walkable: Boolean = true

    @Column(name = "path", nullable = false)
    open var path: String = ""
}